asab/zookeeper/container.py

```diff
@@ -31,22 +30,31 @@ def __init__(self, app, config_section_name, config=None):
 	async def initialize(self, app):
 		await self.ZooKeeper.start()
 		await self.ZooKeeper.ensure_path(self.ZooKeeperPath)
+		await self.do_advertise(self)
 
 	async def finalize(self, app):
 		await self.ZooKeeper.close()
 
-	async def advertise(self, data, path, encoding="utf-8"):
-		if isinstance(data, dict):
-			data = json.dumps(data).encode(encoding)
-		elif isinstance(data, str):
-			data = data.encode(encoding)
-		elif asyncio.iscoroutinefunction(data):
-			data = await data
-		elif callable(data):
-			data = data()
+	async def advertise(self,data, path):
+		self.Data =data
+		self.Path = path
+		self.App.PubSub.subscribe("Application.tick/300!", self.on_tick)
```

Comment:
You should subscribe at `initialize()` since multiple `advertise()` calls can happen.

Here you call `do_advertise()`